odoo/models.py

```diff
@@ -2856,6 +2874,9 @@ def _setup_base(self):
 
         cls._setup_done = True
 
+        for field in cls._fields.values():
+            field.prepare_setup()
```

Comment:
Is the method prepare_setup only here to avoid accessing the _private_ attribute `_setup_done`?

Reply:
I wanted to hide what is actually done in `field` for setup.
This API allows me to change the setup implementation, if I want to.